### PR TITLE
Mailing

### PR DESCRIPTION
**Pull request by Vicctorri**

### CODE REVIEW

File: app/Http/Controllers/ArticleController.php [hunk 1]
```diff
@@ -3,12 +3,13 @@
 namespace App\Http\Controllers;
 
 use Illuminate\Http\Request;
+use App\Models\Article;
```
Comment (egorbwork) [inline app/Http/Controllers/ArticleController.php:6]: This changes, should not be present in this pr.

File: app/Http/Controllers/BlogController.php [hunk 2]
```diff
@@ -4,12 +4,31 @@
...
+use Illuminate\Pagination\Paginator;
+use Illuminate\Pagination\LengthAwarePaginator;
 
 class BlogController extends Controller
 {
     public function index(){
-        $articles = Article::orderBy('created_at', 'DESC')->get()->all();
 
-        return view('blog.blog', ['articles'=>$articles]);
+        $request = request() ->all();
```
Comment (egorbwork) [inline app/Http/Controllers/BlogController.php:15]: And this one too.

File: app/Http/Controllers/ContactController.php [hunk 3]
```diff
@@ -2,12 +2,47 @@
...
 
 class ContactController extends Controller
 {
     public function index(){
         return view('home.contact');
     }
+
+    public function send(ContactRequest $request):RedirectResponse
+    {
+        $callback = function (string $test): string{
```
Comment (egorbwork) [inline app/Http/Controllers/ContactController.php]: Callback was removed in the end of lesson. Please rewatch end of lesson.

File: app/Http/Controllers/ContactController.php [hunk 4]
```diff
@@ -2,12 +2,47 @@
...
+                'lastname' => $data['lastname'],
+                'email' => $data['email'],
+                'phonenumber' => $data['phonenumber'],
+                'subject' => $data['subject'],
+                'messageText' => $data['message'],
+            ],
+        
+        function(Message $message) use ($data){ 
+            $message->subject('mesage from ' . $data['email']);
+            $message->to('tech@baloon.app');
```
Comment (egorbwork) [inline app/Http/Controllers/ContactController.php]: Please adjust `to` and `from` methods to use your project related values.

File: app/Http/Request/ContactRequest.php [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+{
+    public function authorize(): bool
+    {
+        return true;
+    }
+
+    public function rules(): array
+    {
+        return[
+            'firstname' => ['min:2', 'required', 'string'],
```
Comment (egorbwork) [inline app/Http/Request/ContactRequest.php]: I recommend required to be first one, string second and after remaining ones.

**Review by egorbwork — CHANGES_REQUESTED** [review]
Pr containing non related changes is more complicated to review. Only functionality part is not looking bad, but it is still incomplete.

File: app/Http/Controllers/ContactController.php (hunk 4 above)
Comment (Vicctorri) [reply app/Http/Controllers/ContactController.php]: changed

File: app/Http/Request/ContactRequest.php (hunk 5 above)
Comment (Vicctorri) [reply app/Http/Request/ContactRequest.php]: +

**Review by egorbwork — APPROVED** [review]
It can go in this way but I requested to have select in the contact us form. And also exists place to  fix in template. I will allow to fix it in next pr. But it is last chance.

**Pull request merged**